### PR TITLE
Guard analysis-by-hash against file classification tampering

### PR DESCRIPTION
**Pull request by cccs-rs**
Related PR: https://github.com/CybercentreCanada/assemblyline-ui-frontend/pull/325

### CODE REVIEW

File: assemblyline_ui/api/v4/ingest.py [hunk 1]
```diff
@@ -216,19 +217,40 @@ def ingest_single_file(**kwargs):
...
+            "type": "INGEST"
+        })
+
+        # Apply provided params
+        s_params.update(data.get("params", {}))
+
         # Load file
         if not binary:
             if sha256:
+                fileinfo = STORAGE.file.get_if_exists(sha256, as_obj=False)
```
Comment (cccs-sgaron) [inline assemblyline_ui/api/v4/ingest.py:239]: You've forgot to add the archive_access param.

`archive_access=config.datastore.ilm.update_archive`

File: assemblyline_ui/api/v4/ingest.py [hunk 2]
```diff
@@ -216,19 +217,40 @@ def ingest_single_file(**kwargs):
...
                 if FILESTORE.exists(sha256):
-                    # Try to get file info from the DB instead of re-computing it
-                    fileinfo = STORAGE.file.get_if_exists(sha256, as_obj=False,
-                                                          archive_access=config.datastore.ilm.update_archive)
-                    if not fileinfo:
-                        # Could not find the file info in the DB, we will re-compute it
-                        FILESTORE.download(sha256, out_file)
+                    if fileinfo:
+                        if not Classification.is_accessible(user['classification'], fileinfo['classification'],
+                                                            ignore_invalid=True):
```
Comment (cccs-sgaron) [inline assemblyline_ui/api/v4/ingest.py:243]: We should not ignore invalid classification.

File: assemblyline_ui/api/v4/submit.py [hunk 3]
```diff
@@ -329,7 +329,18 @@ def submit(**kwargs):
         extra_meta = {}
         if not binary:
             if sha256:
+                fileinfo = STORAGE.file.get_if_exists(sha256, as_obj=False)
                 if FILESTORE.exists(sha256):
+                    if fileinfo:
+                        if not Classification.is_accessible(user['classification'], fileinfo['classification'],
+                                                            ignore_invalid=True):
```
Comment (cccs-sgaron) [inline assemblyline_ui/api/v4/submit.py:336]: We should not ignore invalid classification

File: assemblyline_ui/api/v4/submit.py [hunk 4]
```diff
@@ -329,7 +329,18 @@ def submit(**kwargs):
         extra_meta = {}
         if not binary:
             if sha256:
+                fileinfo = STORAGE.file.get_if_exists(sha256, as_obj=False)
```
Comment (cccs-sgaron) [inline assemblyline_ui/api/v4/submit.py:332]: You've forgot to add the archive_access param.

`archive_access=config.datastore.ilm.update_archive`

**Review by cccs-sgaron — CHANGES_REQUESTED** [review]

**Review by cccs-sgaron — APPROVED** [review]

**Review by cccs-douglass — APPROVED** [review]

**Pull request merged**